force-app/main/default/classes/ExchangeRateScheduler.cls

```
@@ -0,0 +1,243 @@
+        try {
+            System.debug('Starting scheduled exchange rate update job');
+            
+            // Check if it's a weekday (CNB doesn't update on weekends)
+            if (isWeekday()) {
+                // Start batch job to process exchange rates
+                ExchangeRateScheduler batchJob = new ExchangeRateScheduler();
+                Database.executeBatch(batchJob, 10);
+                
+                System.debug('Exchange rate update batch job started');
+            } else {
+                System.debug('Skipping exchange rate update - not a weekday');
+            }
+            
+        } catch (Exception e) {
+            System.debug(LoggingLevel.ERROR, 'Error in scheduled exchange rate update: ' + e.getMessage());
+            System.debug(LoggingLevel.ERROR, 'Stack trace: ' + e.getStackTraceString());
+            
+            // Log error to custom object or send notification
+            logSchedulerError(e);
+        }
+    }
+    
+    /**
+     * @description Batch start method
+     * @param bc BatchableContext
+     * @return Database.QueryLocator
+     */
+    public Database.QueryLocator start(Database.BatchableContext bc) {
+        // Return empty query locator since we don't need to process existing records
+        return Database.getQueryLocator('SELECT Id FROM Currency__c LIMIT 0');
+    }
+    
+    /**
+     * @description Batch execute method
+     * @param bc BatchableContext
+     * @param scope List of SObject records
+     */
+    public void execute(Database.BatchableContext bc, List<SObject> scope) {
+        try {
+            // Fetch and process exchange rates
+            ExchangeRateProvider.ExchangeRateResult result = ExchangeRateProvider.fetchAndProcessExchangeRates();
+            
+            if (result.success) {
+                System.debug('Exchange rate update completed successfully. Processed: ' + result.processedCount);
+            } else {
```

🛑 **[Robustness] A Failing Job Must Not Report 'Completed'**

This is a dangerous silent failure. If the initial data fetch fails, the batch currently does nothing and finishes gracefully. On the Apex Jobs page, an admin will see a green "Completed" status, but in reality, our core process didn't run at all. We could be operating on stale data for days without anyone noticing.

For a catastrophic failure like this, the job *must* fail. We should throw a custom exception from this `else` block. That will correctly terminate the job and put a big, red "Failed" status in the Apex Jobs monitor, which is the right signal to send when the process is broken.